config/7.1.0/securebanking/ig/scripts/groovy/ProcessPaymentConsent.groovy

```diff
@@ -20,39 +20,74 @@ def method = request.method
 switch(method.toUpperCase()) {
 
     case "POST":
-        paymentIntentData = request.entity.getJson()
-        def tz = TimeZone.getTimeZone("UTC");
-        def df = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'");
-        df.setTimeZone(tz);
-        def nowAsISO = df.format(new Date());
-
         def consentId = routeArgConsentIdPrefix + UUID.randomUUID().toString()
-
-        paymentIntentData._id = consentId
-        paymentIntentData.Data.ConsentId = consentId
-        paymentIntentData.Data.Status = "AwaitingAuthorisation";
-        paymentIntentData.Data.CreationDateTime = nowAsISO
-        paymentIntentData.Data.StatusUpdateDateTime = nowAsISO
-        paymentIntentData.apiClient = [ "_ref" : "managed/" + routeArgObjApiClient + "/" + apiClientId ]
-
-        logger.debug(SCRIPT_NAME + "final json [" + paymentIntentData + "]")
-        request.setEntity(paymentIntentData)
-
-
+        paymentIntentData = request.entity.getJson()
+        processProcessPaymentConsentRequestData(consentId, paymentIntentData)
+
+        def version = getObApiVersion(request)
+        def idmIntent = [
+                _id: consentId,
+                OBVersion: version,
+                OBIntentObjectType: "OBReadConsentResponse1",
```

Comment:
`OBIntentObjectType` hardcode with the same value for all payments doesn't make sense, must be the OB object type json-based, isn't? for example `OBWriteDomesticConsentResponse5` for domestic payment v3.1.8 if we want a relevant value for `OBIntentObjectType`, if not doesn't make sense the field `OBIntentObjectType`.

Reply:
Ooops, good spot!

Reply:
To fix we will add a route argument to enable the OBIntentObjectType to be configured per payment route.

For now, we will hardcode this as the response type supported by OBIE Read/Write specs 3.1.8-3.1.10, this is a temporary solution until we move the processing of consents to the RS. 

Within the RS it will be much simpler to determine the OBIntentObjectType (and OBVersion) values, as these will be known to each ConsentApiController implementation.